cvise/passes/clanghints.py

```diff
@@ -0,0 +1,92 @@
+import json
+import logging
+from pathlib import Path
+import shlex
+import subprocess
+import time
+from typing import Union
+
+from cvise.passes.abstract import BinaryState
+from cvise.passes.hint_based import HintBasedPass, HintState
+
+
+CLANG_STD_CHOICES = ('c++98', 'c++11', 'c++14', 'c++17', 'c++20', 'c++2b')
+
+
+class ClangState(HintState):
+    """Extends HintState to store additional information needed by ClangHintsPass.
+
+    See the comment in ClangHintsPass for the background."""
+
+    def __init__(self, clang_std: str, hint_count: int, hints_file_path: Path, binary_state: BinaryState):
+        super().__init__(hint_count, hints_file_path, binary_state)
+        self.clang_std = clang_std
+
+    @staticmethod
+    def wrap(parent: Union[HintState, None], clang_std: str) -> Union[HintState, None]:
+        if parent is None:
+            return None
+        return ClangState(clang_std, parent.hint_count, parent.hints_file_path, parent.binary_state)
+
+
+class ClangHintsPass(HintBasedPass):
+    """A pass that performs reduction using the hints produced by the clang_delta tool.
+
+    Implementation-wise, we don't use default new/advance/advance_on_success implementation from the base class, because
+    we want to brute-force Clang's `--std=` parameter that maximizes the generated set of hints. This requires having
+    special logic in new() and carrying over some extra information from new() to advance_on_success() throughout all
+    advance() calls.
+    """
+
+    def check_prerequisites(self):
+        return self.check_external_program('clang_delta')
+
+    def new(self, test_case, tmp_dir, **kwargs):
+        # Choose the best standard unless the user provided one.
```

Comment:
A note for reviewing: conceptually this follows the code from `clangbinarysearch.py`.